From 2305ebc528f9e8e62641a20e59912971ec650a5d Mon Sep 17 00:00:00 2001
From: Rishabh Kabra <rkabra@google.com>
Date: Fri, 25 Jun 2021 08:53:00 -0700
Subject: [PATCH 11/20] Update docs copyright year dynamically.

PiperOrigin-RevId: 381480018
---
 docs/conf.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/docs/conf.py b/docs/conf.py
index 1e043f7..1f75426 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -26,6 +26,7 @@
 
 # pylint: disable=g-bad-import-order
 # pylint: disable=g-import-not-at-top
+import datetime
 import inspect
 import os
 import sys
@@ -37,7 +38,7 @@ import tree
 # -- Project information -----------------------------------------------------
 
 project = 'Tree'
-copyright = '2019, DeepMind'  # pylint: disable=redefined-builtin
+copyright = f'{datetime.date.today().year}, DeepMind'  # pylint: disable=redefined-builtin
 author = 'DeepMind'
 
 # -- General configuration ---------------------------------------------------
-- 
2.25.1

